scripts/configurator/parser/cmake_var.py

```diff
@@ -1,7 +1,104 @@
-from typing import Dict, Any, Type, Mapping
+from typing import Dict, Any, Optional, Type, Mapping, List
+import re
+from sys import stderr
+
+
+class metadata:
+   keyword_expr = re.compile(r"([A-Z_]+)")
+
+   def __init__(self, comment: str) -> None:
+      self.depends_on: Optional[List[str]] = None
+      self.group: Optional[str] = None
+      self.realtype: Optional[str] = None
+
+      self.kw_to_parse: Mapping[str, function] = {
+         "REALTYPE" : self.parse_realtype,
+         "DEPENDS_ON" : self.parse_depends_on,
+         "GROUP" : self.parse_group
+      }
+
+      self.kw_to_setter = {
+         "REALTYPE": self.get_setter_realtype,
+         "DEPENDS_ON": self.get_setter_depends_on,
+         "GROUP": self.get_setter_group,
```

Comment:
This is super ugly. Do:

```
      self.kw_to_setter = {
         "REALTYPE": self.get_setter_realtype(),
         "DEPENDS_ON": self.get_setter_depends_on(),
         "GROUP": self.get_setter_group(),
    }
```

you really need to put the **setter closure** in that dict, **not** the getter function for the closure.

Reply:
oh, ok. That actually makes sense